src/Ast/AST.php

```diff
@@ -0,0 +1,90 @@
+<?php declare(strict_types=1);
+
+namespace Google\Generator\Ast;
+
+use Google\Generator\Collections\Vector;
+
+/** Base of the PHP code AST. */
+abstract class AST
+{
+    /**
+     * Create a block of PHP code.
+     * 
+     * @param array $code The code to include in this block.
+     *     Each item must be an AST instance or a Vector thereof.
+     *     Null values will be ignored.
+     * 
+     * @return AST
+     */
+    public static function block(...$code): AST
+    {
+        $code = Vector::new($code)
+            ->flatten()
+            ->filter(fn($x) => !is_null($x));
+        return new class($code) extends AST
```

Comment:
What is the reason for using an anonymous class? Is this because this is more readable than having a different subclass for each block/return/var etc?

Reply:
Exactly; there's going to be quite a few very simple classes here, and in my opinion it's much more readable to keep them all local in this file.
If some particular anonymous classes become too complex, they can be split out without having to split out all of them into separate files.